Dockerfile

```diff
@@ -1,11 +1,5 @@
-FROM golang:1.11-alpine3.8 AS build
+FROM golang:1.12-alpine3.10 AS build
 
-RUN apk add --no-cache curl git build-base && \
-	curl -SL -o /usr/bin/dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64 && \
-	chmod +x /usr/bin/dep
-
-ENV MAINDIR $GOPATH/src/github.com/go-ocf/go-coap
-WORKDIR $MAINDIR
-COPY Gopkg.toml Gopkg.lock ./
-RUN dep ensure -v --vendor-only
```

Comment:
I would like to have this optimization preserved for Go Modules to allow Docker to cache downloaded dependencies when the `go.mod` file is unchanged.

Reply:
Good point. I'll fix that.

Reply:
Hi,

I am not sure the current version will allow Docker to cache the download as copying `.` will generally change quite often invalidating the cache of any steps afterwards. I believe you should just copy `go.mod` and `go.sum` first, do the `go mod download` and then copy the rest. This way if `go.mod` and `go.sum` didn't change, the downloading will still be taken from the cache.

Reply:
```dockerfile
COPY go.mod go.sum ./
RUN go mod download
COPY . .
````

Reply:
@ivajloip @cizmazia good point!

Reply:
Updated.